File: 338. Counting Bits/solution2.py
class Solution(object):
    def countBits(self, num):
        """
        :type num: int
        :rtype: List[int]
        """
        ans = [0]
        for n in range(1,num+1):
            #ans.append(ans[n>>1]+(n&1))
            
            ans.append(ans[n-(1<<int(math.log(n,2)))] + 1)
        return ans

File: Algorithms/292. Nim Game/Solution2.java
//找规律
// n =1,2,3 为 true,n = 4 时，为false
// n = 5,6,7时，通过拿1~3颗，转换为n = 4的情况，即为 true；
// n = 8 时，无论先拿几颗，对手采取相应措施，转换为n = 4的情况，为false

public class Solution {
    public boolean canWinNim(int n) {
        return n%4 > 0;
    }
}